storage/rocksdb/ha_rocksdb.cc

```diff
@@ -9583,6 +9583,36 @@ int ha_rocksdb::update_indexes(const struct update_row_info &row_info,
   return HA_EXIT_SUCCESS;
 }
 
+static void calc_autoinc_difference(const uchar *const old_row,
+                                 const uchar *const new_row,
+                                 TABLE* table,
+                                 ulonglong& auto_inc) {
+  Field* field;
+  const char* o_ptr;
+  const char* n_ptr;
+  uint i;
+
+  auto_inc = 0;
+
+  if (!old_row) {
+    return;
+  }
+
+  for (i = 0; i < table->s->fields; i++) {
```

Comment:
Do we have to do the for loop? Doesn't `table->s->next_number_index` contain the field we want?

Reply:
Not always, for example,

CREATE TABLE t1 (a INT, pk INT AUTO_INCREMENT, PRIMARY KEY(pk)) ENGINE=RocksDB;
INSERT INTO t1 (a) VALUES (20);

(gdb) p table->s->next_number_index
$18 = 0
(gdb) p table->field[0]->val_uint()
$19 = 20

Reply:
Actually, I got confused on what `table->s->next_number_index` meant. It is actually telling us which index the autoincrement is one (not the "index" of the field array). In this case, it's 0 because it is on primary key.

Looking at `ha_rocksdb::load_auto_incr_value_from_index`, we just assume that the first column of the index is the autoincrement field. Looks like something similar is assumed in InnoDB (row_search_autoinc_read_column is always called with col_no == 0).

Reply:
How about the updated patch?